src/accumulation.jl

```diff
@@ -38,30 +38,38 @@ end
     is_inplaceable_destination(x) -> Bool
 
 Returns true if `x` is suitable for for storing inplace accumulation of gradients.
-For arrays this boils down `x .= y` if will work to mutate `x`, if `y` is an appropriate
-tangent.
-Wrapper array types do not need to overload this if they overload `Base.parent`, and are
-`is_inplaceable_destination` if and only if their parent array is.
-Other types should overload this, as it defaults to `false`.
+For arrays this means `x .= y` will mutate `x`, if `y` is an appropriate tangent.
+
+Here "appropriate" means that both are real or both are complex,
+and that for structured matrices like `x isa Diagonal`, `y` shares this structure.
+
+Wrapper array types should overload this function if they can be written into.
+Before ChainRulesCore 1.16, it would guess `true` for most wrappers based on `parent`,
+but this is not safe, e.g. it will lead to an error with ReadOnlyArrays.jl. 
```

Comment:
```suggestion
!!! note "history"
    Wrapper array types should overload this function if they can be written into.
    Before ChainRulesCore 1.16, it would guess `true` for most wrappers based on `parent`,
    but this is not safe, e.g. it will lead to an error with ReadOnlyArrays.jl. 
```